examples/component-webhooks/capabilities/zarf-types.ts

```diff
@@ -0,0 +1,23 @@
+export interface DeployedPackage {
+  componentWebhooks?: { [key: string]: { [key: string]: Webhook } };
+  deployedComponents: DeployedComponent[];
+  generation: number;
+  name: string;
+}
+
+export interface DeployedComponent {
+  images: string[];
+  name: string;
+  observedGeneration: number;
+  status: string;
+}
+
+export interface Webhook {
+  name: string;
+  observedGeneration: number;
+  status: string;
+  /**
+   * The number of seconds that Zarf will wait for the webhook to run before timing out and returning an error.
+   */
+  waitDurationSeconds?: number;
+}
```

Comment:
did we want to generate these similar to how the zarf init aws package works? It may make sense to just add an `internal` command to get types out of zarf more generically and use it in both places (and allow others to use it too)

Reply:
Added a `zarf internal generate-types-schema` command